src/com/googlecode/shavenmaven/Dependencies.java

```diff
@@ -40,37 +43,43 @@ public static Dependencies load(File file, PrintStream out) {
         return new Dependencies(SupportedArtifacts.supportedArtifacts().artifacts(some(file)), out);
     }
 
-    public boolean update(File directory) {
+    public Sequence<Option<String>> update(File directory) {
         files(directory).
                 filter(where(name(), is(not(in(artifacts.map(Artifact::filename))))).and(not(isDirectory()))).
                 map(delete()).realise();
         final Resolver resolver = new Resolver(directory, out);
-        return artifacts.filter(not(Artifact.functions.existsIn(directory))).mapConcurrently(resolve(resolver)).forAll(is(true));
+        return artifacts.filter(not(Artifact.functions.existsIn(directory))).mapConcurrently(resolve(resolver));
```

Comment:
Looking at this code again, I'm not sure the way you have currently added class path support is actually going to work, nor do I think the method signature actually makes sense.

Currently where do you handle building the class path when no dependencies download? Changing 'resolve' to return a class path seems weird as resolve is only called when something has changed.

Generating a class path should really just be completely separate piece of code with no code change to the existing API. I'd just have a new Class that iterates over the Artifacts.

Sorry I didn't catch this right at the start.